core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -274,7 +265,7 @@ private[spark] object ClosureCleaner extends Logging {
       // required fields from the original object. We need the parent here because the Java
       // language specification requires the first constructor parameter of any closure to be
       // its enclosing object.
-      val clone = instantiateClass(cls, parent, inInterpreter)
+      val clone = instantiateClass(cls, parent, Utils.isInInterpreter)
```

Comment:
Very valid point. I overlooked this being more focused on clarifying the confusion around exception handling and negation. Will fix.